tests/StackExchange.Redis.Tests/Keys.cs

```diff
@@ -264,5 +264,25 @@ public async Task TouchIdleTimeAsync()
                 Assert.True(idleTime1 < idleTime);
             }
         }
+
+        [Fact]
+        public async Task KeyEncoding()
+        {
+            using var muxer = Create();
+            var key = Me();
+            var db = muxer.GetDatabase();
+
+            db.KeyDelete(key, CommandFlags.FireAndForget);
+            db.StringSet(key, "new value", flags: CommandFlags.FireAndForget);
+
+            Assert.Equal("embstr", db.KeyEncoding(key));
+            Assert.Equal("embstr", await db.KeyEncodingAsync(key));
+
+            db.KeyDelete(key, CommandFlags.FireAndForget);
+            db.ListLeftPush(key, "new value", flags: CommandFlags.FireAndForget);
+
+            Assert.Equal("quicklist", db.KeyEncoding(key));
+            Assert.Equal("quicklist", await db.KeyEncodingAsync(key));
```

Comment:
This will vary by server version as to when it chooses to ziplist, recommend using something like
```suggestion
            // Depending on server version, this is going to vary - we're sanity checking here.
            var listTypes = new [] { "ziplist", "quicklist" };
            Assert.Contains(db.KeyEncoding(key), listTypes);
            Assert.Contains(await db.KeyEncodingAsync(key), listTypes);
```
(note: this assumes string changes above)